### PR TITLE
Add Thrift Support for ClusterStatsResource

### PR DESCRIPTION
**Pull request by derek-xia**
Adds thrift serde support for the getClusterStats endpoint.

Test plan - Unit tests

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by derek-xia** [comment]
@ajaygeorge Picked this up as a bootcamp task, am I on the right track with the unit tests?

**Comment by ajaygeorge** [comment]
@derek-xia Yes. Let me know if this PR is ready to be reviewed.

**Comment by derek-xia** [comment]
I think this is ready for review.

File: presto-tests/src/test/java/com/facebook/presto/server/TestClusterStatsResource.java [hunk 1]
```diff
@@ -101,15 +111,23 @@ public void testGetClusterStats()
...
                 .setHeader(PRESTO_USER, "user")
                 .setUri(uriBuilderFrom(server.getBaseUrl().resolve("/v1/cluster")).build())
-                .setFollowRedirects(followRedirects)
-                .build();
-
-        return client.execute(request, createJsonResponseHandler(jsonCodec(ClusterStatsResource.ClusterStats.class)));
+                .setFollowRedirects(followRedirects);
+        if (useThriftEncoding) {
+            Request request = builder
+                    .setHeader("Accept", ThriftRequestUtils.APPLICATION_THRIFT_COMPACT)
```
Comment (ajaygeorge) [inline presto-tests/src/test/java/com/facebook/presto/server/TestClusterStatsResource.java:122]: you can use `ThriftRequestUtils.prepareThriftGet` which gives you a builder.
Also this just tests for one protocol and misses out on `APPLICATION_THRIFT_BINARY` and `APPLICATION_THRIFT_FB_COMPACT` . You can create enhance the

Comment (ajaygeorge) [inline presto-tests/src/test/java/com/facebook/presto/server/TestClusterStatsResource.java:122]: you can use `ThriftRequestUtils.prepareThriftGet` which gives you a builder.
Also this just tests for one protocol and misses out on `APPLICATION_THRIFT_BINARY` and `APPLICATION_THRIFT_FB_COMPACT` . You can create enhance the data provider to add these types as well.

**Review by ajaygeorge — CHANGES_REQUESTED** [review]

File: presto-tests/src/test/java/com/facebook/presto/server/TestClusterStatsResource.java [hunk 2]
```diff
@@ -101,15 +112,21 @@ public void testGetClusterStats()
...
+        Request request = builder
                 .setHeader(PRESTO_USER, "user")
                 .setUri(uriBuilderFrom(server.getBaseUrl().resolve("/v1/cluster")).build())
                 .setFollowRedirects(followRedirects)
                 .build();
-
-        return client.execute(request, createJsonResponseHandler(jsonCodec(ClusterStatsResource.ClusterStats.class)));
+        if (useThriftEncoding) {
+            ThriftCodecManager codecManager = new ThriftCodecManager();
+            return client.execute(request, new ThriftResponseHandler<>(codecManager.getCodec(ClusterStatsResource.ClusterStats.class))).getValue();
```
Comment (ajaygeorge) [inline presto-tests/src/test/java/com/facebook/presto/server/TestClusterStatsResource.java:125]: static import for ClusterStatsResource.ClusterStats.class

**Review by ajaygeorge — APPROVED** [review]
LGTM % a comment

**Comment by derek-xia** [comment]
Addressed the comment

**Review by rschlussel — APPROVED** [review]

**Pull request merged**